extensions/local-storage.js

```diff
@@ -5,29 +5,38 @@
 (function (Scratch) {
   "use strict";
 
+  const vm = Scratch.vm;
+  const runtime = vm.runtime;
+
   if (!Scratch.extensions.unsandboxed) {
     throw new Error("Local Storage must be run unsandboxed");
   }
 
   const PREFIX = "extensions.turbowarp.org/local-storage:";
-  let namespace = "";
-  const getFullStorageKey = () => `${PREFIX}${namespace}`;
 
-  let lastNamespaceWarning = 0;
+  const soup_ =
+    "!#%()*+,-./:;=?@[]^_`{|}~" +
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
 
-  const validNamespace = () => {
-    const valid = !!namespace;
-    if (!valid && Date.now() - lastNamespaceWarning > 3000) {
-      alert(
-        Scratch.translate(
-          'Local Storage extension: project must run the "set storage namespace ID" block before it can use other blocks'
-        )
-      );
-      lastNamespaceWarning = Date.now();
+  const uid = function () {
+    const length = 10;
+    const soupLength = soup_.length;
+    const id = [];
+    for (let i = 0; i < length; i++) {
+      id[i] = soup_.charAt(Math.random() * soupLength);
     }
-    return valid;
+    return id.join("");
   };
 
+  if (!runtime.extensionStorage["localstorage"]) {
```

Comment:
think you're gonna have the same problem as the discord extension here

Reply:
evidently there should probably be a simpler api for common usecases like this

Reply:
Good call